packages/scheduler/src/Scheduler.js

```diff
@@ -160,10 +186,23 @@ function flushWork(hasTimeRemaining, initialTime) {
       }
       return false;
     }
+  } catch (error) {
+    if (enableProfiling) {
+      if (currentTask !== null) {
+        const currentTime = getCurrentTime();
+        markTaskErrored(currentTask, currentTime);
+        currentTask.isQueued = false;
+      }
+    }
+    throw error;
```

Comment:
It's kind of non-ideal that in prod path we now just have `catch (err) { throw error; }`. I could restructure to make it profile-only but didn't bother.

Reply:
Yea this is not great. Does this cause any deopts that we know of? Would be an extremely bad place to deopt.

Let's just hoist this body out into a separate function that is either wrapped in catch or not. It should get inlined by closure, however, maybe we don't actually want it to inline since that limits the deopt to the outer function in those older VMs.